### PR TITLE
Features

### PR DESCRIPTION
**Pull request by omokaro-faith**

### CODE REVIEW

File: .coveralls.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+repo_token: wc2daTtJVXyxbgbn9C3K4cnGO67TBffSW
```
Comment (olawalejarvis) [inline .coveralls.yml:1]: It's not advisable to push this file, as it contains your repo token which should be confidential.
repo token is only useful when you want to report coverage to coveralls directly from your local environment.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+language: node_js
+node_js:
+ - stable
+ - 6
```
Comment (olawalejarvis) [inline .travis.yml:4]: How are you running your test on travis?

File: Model/app.js [hunk 3]
```diff
@@ -36,7 +36,8 @@ app.controller('myController', ['$scope', ($scope) => {
       }
       $scope.documents = $scope.fileNames.push(file[i].name);
       $scope.showUploaded = true;
-      $scope.$apply($scope.documents);
+      // $scope.$apply($scope.documents);
```
Comment (olawalejarvis) [inline Model/app.js:39]: It advisable to remove all commented codes from code base

File: Model/app.js [hunk 4]
```diff
@@ -46,45 +47,59 @@ app.controller('myController', ['$scope', ($scope) => {
       };
       reader.readAsText(file[i]);
     }
+    console.log("++++++++upload", $scope.fileContent)
```
Comment (olawalejarvis) [inline Model/app.js:50]: It's not advisable to leave any console.log() in code base

File: Model/inverted-index.js [hunk 5]
```diff
@@ -2,24 +2,17 @@ class Index {
...
-  //   }
-  //   return true;
-  // }
 
   tokenize(words) {
-    const word = words.toLowerCase();
-    return word.trim().replace(/[^a-zA-Z 0-9]+/g, '');
+    const token = words.replace(/,+/g, ' ')
+    .replace(/[^a-zA-Z 0-9\s]+/g, '')
+    .replace(/\s\s/g, ' ')
```
Comment (olawalejarvis) [inline Model/inverted-index.js:12]: Great work, but I'm curious why are you using three different regex replace() here?

Comment (omokaro-faith) [reply Model/inverted-index.js:12]: I have made changes to the tokenize method, I just couldn't find a better way of replacing extra spaces, comma and non-alphanumeric characters

File: Model/app.js (hunk 4 above)
Comment (omokaro-faith) [reply Model/app.js:50]: Thanks I have made changes to the file by removing the console.log() in code base

File: Model/app.js (hunk 3 above)
Comment (omokaro-faith) [reply Model/app.js:39]: Thanks I have made changes to the file by removing unwanted comments

File: Model/app.js [hunk 6]
```diff
@@ -46,45 +47,59 @@ app.controller('myController', ['$scope', ($scope) => {
...
-    fileContent = JSON.parse(fileContent);
-    $scope.fileContent[fileName] = fileContent;
-    const length = fileContent.length;
-    const temp = [];
-    for (let i = 0; i < length; i++) {
-      temp.push(i);
+      $scope.numberOfDocuments[fileName] = temp;
+      $scope.filesToSearch.push(fileName);
+      $scope.showTable = true;
+      console.log($scope.indices);
```
Comment (fortunee) [inline Model/app.js:80]: Remove all `console.log`. Don't leave logs in your code!

File: Model/inverted-index.js [hunk 7]
```diff
@@ -34,9 +27,10 @@ class Index {
       status: true,
       msg: 'Valid File',
     };
-
+    // console.log('my type', typeof file);
```
Comment (fortunee) [inline Model/inverted-index.js:30]: Remove the commented out code here and every other recurrence of it throughout your code.

File: Model/inverted-index.js [hunk 8]
```diff
@@ -113,28 +103,44 @@ class Index {
     }
   }
 
-  searchIndex(terms, fileName) {
-    const selectedIndex = this.getIndex(fileName);
+  searchIndex(sTerms, fileName) {
```
Comment (fortunee) [inline Model/inverted-index.js:106]: Use descriptive names for variables and parameters, I'm able to understand that `sTerms` means `searchTerms` cos I'm quite familiar with the program but what happens to a strange developer reading your code?.

File: spec/index-spec.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+},
+
+{
+  title: 'Checkpoint Two',
+  text: 'Coming soon',
+},
+];
+
+
+// invertedIndex.createIndex('feedback.json', work);
```
Comment (fortunee) [inline spec/index-spec.js:21]: Remove commented out code

File: spec/index-spec.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+// Test Suite
+
+
```
Comment (fortunee) [inline spec/index-spec.js:25]: Use single line spaces here instead of double line spaces

File: spec/index-spec.js [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+        of: [1],
+        ants: [1],
+        bullying: [1],
+        humans: [1],
+      });
+    });
+});
+
+
+// describe('Search Index', () => {
```
Comment (fortunee) [inline spec/index-spec.js:91]: Remove all commented out code

**Pull request closed without merging**